src/main/java/core/basesyntax/service/impl/TransactionProcessorServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.TransactionProcessorService;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+import java.util.Map;
+
+public class TransactionProcessorServiceImpl implements TransactionProcessorService {
+    private Map<String, Integer> fruitCounts;
+    private final Map<FruitTransaction.Operation, OperationStrategy> strategies;
+
+    public TransactionProcessorServiceImpl(Map<FruitTransaction.Operation,
+            OperationStrategy> strategies) {
+        this.strategies = strategies;
+    }
+
+    @Override
+    public Map<String, Integer> processTransaction(List<FruitTransaction> transactions) {
```

Comment:
```suggestion
    public Map<String, Integer> processTransactions(List<FruitTransaction> transactions) {
```
name is confusing, given that it accepts list